lib/identity/hostdata/s3.rb

```diff
@@ -16,33 +16,53 @@ def initialize(bucket:, env:, region:, s3_client: nil, logger: nil)
         @s3_client = s3_client
       end
 
-      def download_configs(configs)
-        configs.each do |s3_path, local_path|
-          download_config(s3_path, local_path)
-        end
-      end
+      def download_file(s3_path, local_path)
+        key = build_key(s3_path)
 
-      private
+        logger && logger.info("#{self.class}: downloading s3://#{bucket}/#{key} to #{local_path}")
 
-      def s3_client
-        @s3_client ||= Aws::S3::Client.new(
-          region: region,
-          http_open_timeout: 5,
-          http_read_timeout: 5
-        )
+        s3_response = get_s3_object(key)
+        stream_s3_object_to_file(s3_response, local_path)
       end
 
-      def download_config(s3_path, local_path)
-        FileUtils.mkdir_p(File.dirname(local_path))
+      def read_file(s3_path)
+        key = build_key(s3_path)
 
-        key = format(s3_path, env: env).sub(%r|\A/|, '')
+        logger && logger.info("#{self.class}: reading s3://#{bucket}/#{key}")
 
-        logger && logger.info("#{self.class}: downloading s3://#{bucket}/#{key} to #{local_path}")
+        get_s3_object(key).body.read
+      rescue Aws::S3::Errors::NoSuchKey
+        nil
+      end
+
+      private
 
+      def build_key(s3_path)
+        format(s3_path, env: env).sub(%r|\A/|, '')
+      end
+
+      def get_s3_object(key)
         s3_client.get_object(
           bucket: bucket,
           key: key,
-          response_target: local_path
+        )
+      end
+
+      def stream_s3_object_to_file(s3_object_response, file_path)
+        FileUtils.mkdir_p(File.dirname(file_path))
+
+        File.open(file_path, 'wb') do |file|
+          while bytes = s3_object_response.body.read(512)
+            file.write(bytes)
+          end
+        end
+      end
```

Comment:
This seems to me like the kind of thing we should not re-implement from the S3 gem

As we discussed in Slack, if we need to be able to remove FakeFS to inherit this behavior from the gem, let's remove FakeFS